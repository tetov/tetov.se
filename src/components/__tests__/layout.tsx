import React from "react";
import renderer from "react-test-renderer";
import Layout from "src/components/layout";

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Layout
          footerChildren={<a href="/">Test</a>}
          subHeading="Test heading"
          pathname="/"
        >
          Some content
        </Layout>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
