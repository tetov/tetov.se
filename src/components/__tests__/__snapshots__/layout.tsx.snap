// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="global-wrapper min-h-screen bg-light-bg dark:bg-dark-bg print:bg-light-bg text-light-text dark:text-dark-text print:text-light-text"
>
  <nav
    className="text-center md:justify-between mb-8 border-b-2 border-purple print:hidden"
    data-headlessui-state=""
  >
    <div
      className="mx-auto px-6 sm:px-6 lg:px-8"
    >
      <div
        className="relative flex h-16 items-center justify-start gap-x-6"
      >
        <div
          className="flex flex-shrink-0"
        >
          <a
            href="/"
          >
            <test-file-stub
              className="h-8 w-auto flex flex-shrink-0"
            />
          </a>
        </div>
        <div
          className="flex items-start "
        >
          <a
            href="/"
          >
            <span
              className="text-xl font-bold tracking-tighter leading-tight whitespace-nowrap"
            >
              IÃ°unn Gunda
            </span>
          </a>
        </div>
        <button
          aria-expanded={false}
          className="sm:hidden ml-auto rounded-md p-2 text-gray-400 hover:text-purple text-opacity-100"
          data-headlessui-state=""
          id="headlessui-disclosure-button-:r0:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Open navigation menu
          </span>
          <div
            className="block w-5 left-1/2 top-1/2   transform  -translate-x-1/2 -translate-y-1/2"
          >
            <span
              aria-hidden="true"
              className="block absolute h-0.5 w-5 bg-current transform transition duration-500 ease-in-out -translate-y-1.5"
            />
            <span
              aria-hidden="true"
              className="block absolute  h-0.5 w-5 bg-current   transform transition duration-500 ease-in-out  "
            />
            <span
              aria-hidden="true"
              className="block absolute  h-0.5 w-5 bg-current transform  transition duration-500 ease-in-out translate-y-1.5"
            />
          </div>
        </button>
        <div
          className="hidden sm:block ml-auto"
        >
          <div
            className="flex space-x-4"
          >
            <a
              aria-current="page"
              className="bg-purple hover:bg-transparent px-3 py-2 font-medium link-style-alt text-base"
              href="/"
            >
              Projects
            </a>
            <a
              className="px-3 py-2 font-medium link-style-alt text-base"
              href="/posts"
            >
              Posts
            </a>
            <a
              className="px-3 py-2 font-medium link-style-alt text-base"
              href="/about"
            >
              About
            </a>
            <a
              className="px-3 py-2 font-medium link-style-alt text-base"
              href="/contact"
            >
              Contact
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <main
    className="w-full md:w-3/4 mx-auto px-5"
  >
    Some content
  </main>
  <footer
    className="pt-8 pb-16 items-center"
  >
    <a
      href="/"
    >
      Test
    </a>
  </footer>
</div>
`;
